src/test/java/build/buildfarm/worker/resources/ResourceDeciderTest.java

```diff
@@ -666,9 +687,128 @@ public void decideResourceLimitationsTestWorkerName() throws Exception {
   }
+
+  // Function under test: decideResourceLimitations
+  // Reason for testing: Sandbox is not selected based on sandbox settings.
+  // Failure explanation: The sandbox should be off by default.
+  @Test
+  public void decideResourceLimitationsSanboxOffDefault() throws Exception {
+    // ARRANGE
+    Command command = Command.newBuilder().build();
+
+    // ACT
+    ResourceLimits limits =
+        ResourceDecider.decideResourceLimitations(
+            command,
+            "foo",
+            /* defaultMaxCores=*/ 0,
+            /* onlyMulticoreTests=*/ false,
+            /* limitGlobalExecution=*/ false,
+            /* executeStageWidth=*/ 100,
+            /* allowBringYourOwnContainer=*/ false,
+            new SandboxSettings());
+
+    // ASSERT
+    assertThat(limits.useLinuxSandbox).isFalse();
+  }
+
+  // Function under test: decideResourceLimitations
+  // Reason for testing: Sandbox is selected based on sandbox settings.
+  // Failure explanation: The sandbox should have been selected.
+  @Test
+  public void decideResourceLimitationsAlwaysUseSandbox() throws Exception {
+    // ARRANGE
+    Command command = Command.newBuilder().build();
+    SandboxSettings sandboxSettings = new SandboxSettings();
+    sandboxSettings.alwaysUse = true;
+
+    // ACT
+    ResourceLimits limits =
+        ResourceDecider.decideResourceLimitations(
+            command,
+            "foo",
+            /* defaultMaxCores=*/ 0,
+            /* onlyMulticoreTests=*/ false,
+            /* limitGlobalExecution=*/ false,
+            /* executeStageWidth=*/ 100,
+            /* allowBringYourOwnContainer=*/ false,
+            sandboxSettings);
+
+    // ASSERT
+    assertThat(limits.useLinuxSandbox).isTrue();
+  }
+
+  // Function under test: decideResourceLimitations
+  // Reason for testing: Sandbox is selected based on sandbox settings.
+  // Failure explanation: The sandbox should have been selected.
+  @Test
+  public void decideResourceLimitationsSandboxChosenViaBlockNetwork() throws Exception {
+    // ARRANGE
+    Command command =
+        Command.newBuilder()
+            .setPlatform(
+                Platform.newBuilder()
+                    .addProperties(
+                        Platform.Property.newBuilder()
+                            .setName(ExecutionProperties.BLOCK_NETWORK)
+                            .setValue("true")))
+            .build();
+    SandboxSettings sandboxSettings = new SandboxSettings();
+    sandboxSettings.selectForBlockNetwork = true;
+
+    // ACT
+    ResourceLimits limits =
+        ResourceDecider.decideResourceLimitations(
+            command,
+            "foo",
+            /* defaultMaxCores=*/ 0,
+            /* onlyMulticoreTests=*/ false,
+            /* limitGlobalExecution=*/ false,
+            /* executeStageWidth=*/ 100,
+            /* allowBringYourOwnContainer=*/ false,
+            sandboxSettings);
+
+    // ASSERT
+    assertThat(limits.useLinuxSandbox).isTrue();
```

Comment:
I want to see a negative test for this as well, where useLinuxSandbox is false for selectForBlockNetwork is true for non-network blocking platform